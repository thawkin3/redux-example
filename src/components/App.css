.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-body {
  margin: 30px;
}

.App-footer {
  margin: 30px;
  text-align: right;
  position: fixed;
  bottom: 0;
  right: 0;
}
.App-footer a {
  color: #222;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
