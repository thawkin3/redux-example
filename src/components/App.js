import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import Button from './Button';
import CounterResults from './CounterResults';
import logo from '../assets/logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">React+Redux Example</h1>
        </header>
        <div className="App-body">
          <p>This is a basic React+Redux example built using create-react-app. It can be used as a walking skeleton for your base React+Redux app which you can then build out. It contains all the basic boilerplate code for action creators, action types, reducers, and initial state.</p>
          <Button clickHandler={this.props.incrementCounter} buttonText="Increment Counter" />
          <Button clickHandler={this.props.decrementCounter} buttonText="Decrement Counter" />
          <Button clickHandler={this.props.resetCounter} buttonText="Reset Counter" />
          <CounterResults />
        </div>
        <div className="App-footer">
          <a href="https://github.com/thawkin3/redux-example" target="_blank" rel="noopener noreferrer">https://github.com/thawkin3/redux-example</a>
        </div>
      </div>
    );
  }
}

export default connect(null, actions)(App);
