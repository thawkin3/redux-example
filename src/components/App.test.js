import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { App } from './App';
import INITIAL_STATE from '../reducers/initialState.js';


let wrapper;

describe('App', () => {
	beforeEach(() => {
		wrapper = shallow(<App />);
	});

	it('renders a single app', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('renders three Button components', () => {
		expect(wrapper.find('Connect(Button)').length).toEqual(3);
	});

	it('renders CounterResults', () => {
		expect(wrapper.find('Connect(CounterResults)').length).toEqual(1);
	});

	it('renders an app header', () => {
		expect(wrapper.find('.App-header').length).toEqual(1);
	});

	it('matchs snapshot', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});

	it('renders without crashing', () => {
	    const mockStore = configureStore();
	    const store = mockStore(INITIAL_STATE);
	    const div = document.createElement('div');
	    ReactDOM.render(<Provider store={store}><App /></Provider>, div);
	    ReactDOM.unmountComponentAtNode(div);
	});


});	