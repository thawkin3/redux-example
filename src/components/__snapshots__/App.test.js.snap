// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      React+Redux Example
    </h1>
  </header>
  <div
    className="App-body"
  >
    <p
      className="introText"
    >
      This is a basic React+Redux example built using create-react-app. It can be used as a walking skeleton for your base React+Redux app which you can then build out. It contains all the basic boilerplate code for action creators, action types, reducers, and initial state.
    </p>
    <Connect(Button)
      buttonText="Increment Counter"
    />
    <Connect(Button)
      buttonText="Decrement Counter"
    />
    <Connect(Button)
      buttonText="Reset Counter"
    />
    <Connect(CounterResults) />
  </div>
  <div
    className="App-footer"
  >
    <a
      href="https://github.com/thawkin3/redux-example"
      rel="noopener noreferrer"
      target="_blank"
    >
      https://github.com/thawkin3/redux-example
    </a>
  </div>
</div>
`;
